### PR TITLE
sys/vfs: add force option to vfs_umount()

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Sometimes we *need* to unmount the file-system *now* even if there are still open files (e.g. to perform a reboot).

So far this was prevented by a check in `vfs_umount()` to not allow to unmount a fs if it still has open files.

This adds a `force` option that will ignore the check and unmount the fs anyway.

### CODE REVIEW

File: sys/vfs/vfs.c [hunk 1]
```diff
@@ -564,7 +564,7 @@ int vfs_umount(vfs_mount_t *mountp)
         return -EINVAL;
     }
     DEBUG("vfs_umount: -> \"%s\" open=%d\n", mountp->mount_point, atomic_load(&mountp->open_files));
-    if (atomic_load(&mountp->open_files) > 0) {
+    if (atomic_load(&mountp->open_files) > 0 || force) {
```
Comment (miri64) [inline sys/vfs/vfs.c:567]: Shouldn't this be
```suggestion
    if (atomic_load(&mountp->open_files) > 0 && !force) {
```

Maybe an `-EBUSY` test case would be good to properly test this ;-)

**Review by miri64 — APPROVED** [review]
With my new test, I think this can be ACK'd.

**Comment by benpicco** [comment]
Thank you for providing the test!

**Comment by miri64** [comment]
bors merge

**Pull request merged**